include/deal.II/distributed/tria_base.h

```diff
@@ -398,6 +398,455 @@ namespace parallel
+     * function does) because the classes that eventually want the data
+     * back may need to do some setup between the point in time where the
+     * mesh has been recreated and when the data can actually be received.
+     * An example is the parallel::distributed::SolutionTransfer class
+     * that can really only receive the data once not only the mesh is
+     * completely available again on the current processor, but only
+     * after a DoFHandler has been reinitialized and distributed
+     * degrees of freedom. In other words, there is typically a significant
+     * amount of set up that needs to happen in user space before the classes
+     * that can receive data attached to cell are ready to actually do so.
+     * When they are, they use the current function to tell the triangulation
+     * object that now is the time when they are ready by calling the
+     * current function.
+     *
+     * The supplied callback function is then called for each newly locally
+     * owned cell. The first argument to the callback is an iterator that
+     * designates the cell; the second argument indicates the status of the
+     * cell in question; and the third argument localizes a memory area by
+     * two iterators that contains the data that was previously saved from
+     * the callback provided to register_data_attach().
+     *
+     * The CellStatus will indicate if the cell was refined, coarsened, or
+     * persisted unchanged. The @p cell_iterator argument to the callback
+     * will then either be an active,
+     * locally owned cell (if the cell was not refined), or the immediate
+     * parent if it was refined during execute_coarsening_and_refinement().
+     * Therefore, contrary to during register_data_attach(), you can now
+     * access the children if the status is `CELL_REFINE` but no longer for
+     * callbacks with status `CELL_COARSEN`.
+     *
+     * The first argument to this function, `handle`, corresponds to
+     * the return value of register_data_attach(). (The precise
+     * meaning of what the numeric value of this handle is supposed
+     * to represent is neither important, nor should you try to use
+     * it for anything other than transmit information between a
+     * call to register_data_attach() to the corresponding call to
+     * notify_ready_to_unpack().)
+     */
+    void
+    notify_ready_to_unpack(
+      const unsigned int handle,
+      const std::function<
+        void(const cell_iterator &,
+             const CellStatus,
+             const boost::iterator_range<std::vector<char>::const_iterator> &)>
+        &unpack_callback);
+
+    /**
+     * Go through all cells and store the relations between locally
+     * owned quadrants and cells in the private member
+     * local_cell_relations.
+     *
+     * The stored vector will be ordered by the occurrence of quadrants.
+     */
+    virtual void
+    update_cell_relations() = 0;
```

Comment:
This function and all the following until line 849 should be `protected`. You do not want to expose this to the public interface, even though this is an abstract class.